price/api.go

```diff
@@ -1,56 +1,140 @@
-			Current: &Prices{
-				PricePerGiB:  new(big.Int).Div(pricePerGiBMax, big.NewInt(2)),
-				PricePerHour: new(big.Int).Div(pricePerHourMax, big.NewInt(2)),
-				ValidUntil:   time.Now().Add(time.Hour).UTC(),
-			},
-			Previous: &Prices{
-				PricePerGiB:  new(big.Int).Div(pricePerGiBMax, big.NewInt(2)),
-				PricePerHour: new(big.Int).Div(pricePerHourMax, big.NewInt(2)),
-				ValidUntil:   time.Now().Add(time.Minute * 30 * -1).UTC(),
-			},
-		},
+		pricer:    pricer,
+		cfger:     cfger,
+		jwtSecret: jwtSecret,
 	}
 }
 
+// LatestPrices returns latest prices
+// @Summary Latest Prices
+// @Description Latest Prices
+// @Product json
+// @Success 200 {array} pricing.LatestPrices
+// @Router /prices [get]
+// @Tags prices
 func (a *API) LatestPrices(c *gin.Context) {
-	c.JSON(200, a.latestPrices)
+	c.JSON(200, a.pricer.GetPrices())
 }
 
-// LatestPrices holds two sets of prices. The Previous should be used in case
-// a race condition between obtaining prices by Consumer and Provider
-// upon agreement
-type LatestPrices struct {
-	Current  *Prices `json:"current"`
-	Previous *Prices `json:"previous"`
+// GetConfig returns the base pricing config
+// @Summary Price config
+// @Description price config
+// @Product json
+// @Success 200 {array} pricing.Config
+// @Router /prices/config [get]
+// @Tags prices
+func (a *API) GetConfig(c *gin.Context) {
+	cfg, err := a.cfger.Get()
+	if err != nil {
+		log.Err(err).Msg("Failed to get config")
+		c.JSON(http.StatusBadRequest, gorest.NewErrResponse(err.Error()))
+		return
+	}
+	c.JSON(http.StatusOK, cfg)
 }
 
-type Prices struct {
-	PricePerHour *big.Int  `json:"price_per_hour"`
-	PricePerGiB  *big.Int  `json:"price_per_gib"`
-	ValidUntil   time.Time `json:"valid_until"`
+// UpdateConfig updates the pricing config
+// @Summary update price config
+// @Description update price config
+// @Product json
+// @Success 202
+// @Param config body pricing.Config true "config object"
+// @Router /prices/config [post]
+// @Tags prices
+func (a *API) UpdateConfig(c *gin.Context) {
+	var cfg pricing.Config
+	if err := c.BindJSON(&cfg); err != nil {
+		c.JSON(http.StatusBadRequest, gorest.NewErrResponse(err.Error()))
+		return
+	}
+
+	err := a.cfger.Update(cfg)
+	if err != nil {
+		log.Err(err).Msg("Failed to update config")
+		c.JSON(http.StatusBadRequest, gorest.NewErrResponse(err.Error()))
+		return
+	}
+
+	c.Data(http.StatusAccepted, gin.MIMEJSON, nil)
 }
 
 func (a *API) RegisterRoutes(r gin.IRoutes) {
+	r.GET("/prices/config", JWTAuthorized(a.jwtSecret), a.GetConfig)
+	r.POST("/prices/config", JWTAuthorized(a.jwtSecret), a.UpdateConfig)
 	r.GET("/prices", a.LatestPrices)
 }
+
+func JWTAuthorized(secret string) func(*gin.Context) {
+	return func(c *gin.Context) {
+		authHeader := strings.Split(c.Request.Header.Get("Authorization"), "Bearer ")
+		if len(authHeader) != 2 {
+			c.AbortWithStatusJSON(
+				http.StatusUnauthorized,
+				map[string]string{
+					"error": "Malformed Token",
```

Comment:
I guess we aren't breaking the world here, but the error responses will differ depending on if you git an error in this middleware of an actual endpoint. Body here is:
```
map[string]string{"error": "Unauthorized",},
```

Body in the handlers themselves for an error is:
```
Errors []Err `json:"errors"`
```